src/test/java/org/openrewrite/java/migrate/lang/IfElseIfConstructToSwitchTest.java

```diff
@@ -0,0 +1,496 @@
+                          formatted = "null";
+                      }
+                      if (obj instanceof Integer i)
+                          formatted = String.format("int %d", i);
+                      else if (obj instanceof Long l) {
+                          formatted = String.format("long %d", l);
+                      } else if (obj instanceof Double d) {
+                          formatted = String.format("double %f", d);
+                      } else if (obj instanceof String s) {
+                          String str = "String";
+                          formatted = String.format("%s %s", str, s);
+                      } else {
+                          formatted = "unknown";
+                      }
+                      return formatted;
+                  }
+              }
+              """,
+            """
+              class Test {
+                  static String formatter(Object obj) {
+                      String formatted = "initialValue";
+                      if (obj == null) {
+                          formatted = "null";
+                      }
+                      switch (obj) {
+                          case Integer i -> formatted = String.format("int %d", i);
+                          case Long l -> formatted = String.format("long %d", l);
+                          case Double d -> formatted = String.format("double %f", d);
+                          case String s -> {
+                              String str = "String";
+                              formatted = String.format("%s %s", str, s);
+                          }
+                          default -> formatted = "unknown";
+                      }
+                      return formatted;
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void defaultSwitchBlockWithNullCheck() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class Test {
+                  static String formatter(Object obj) {
+                      String formatted = "initialValue";
+                      if (obj == null) {
+                          formatted = "null";
+                      } else if (obj instanceof Integer i)
+                          formatted = String.format("int %d", i);
+                      else if (obj instanceof Long l) {
+                          formatted = String.format("long %d", l);
+                      } else if (obj instanceof Double d) {
+                          formatted = String.format("double %f", d);
+                      } else if (obj instanceof String s) {
+                          String str = "String";
+                          formatted = String.format("%s %s", str, s);
+                      }
+                      return formatted;
+                  }
+              }
+              """,
+            """
+              class Test {
+                  static String formatter(Object obj) {
```

Comment:
This output is incorrect, as the code will no longer compile:

![image](https://github.com/user-attachments/assets/f55b9288-97e8-4515-89c6-6b4e83f07cbe)

Same thing does also apply to some test below this one.